core/dbt/exceptions.py

```diff
@@ -2187,6 +2187,25 @@ def get_message(self) -> str:
         return msg
 
 
+class DuplicateVersionedUnversionedError(ParsingError):
+    def __init__(self, versioned_node, unversioned_node):
+        self.versioned_node = versioned_node
+        self.unversioned_node = unversioned_node
+        super().__init__(msg=self.get_message())
+
+    def get_message(self) -> str:
+        msg = f"""
+dbt found versioned and unversioned models with the name "{self.versioned_node.name}".
+
+Since these resources have the same name, dbt will be unable to find the correct resource
+when looking for ref({self.versioned_node.name}).
```

Comment:
tiny nit - adding quotes around the node: `ref('{self.versioned_node.name}')`

Reply:
Added quotes.